pyfstat/core.py

```diff
@@ -1636,38 +1659,34 @@ def get_semicoherent_twoF(
         #                                       self.BSGLSetup)
         #    return log10_BSGL/np.log10(np.exp(1))
 
-        detStat = 0
+        det_stat_per_segment = self._get_per_segment_det_stat()
+        detStat = det_stat_per_segment.sum()
         if record_segments:
-            self.detStat_per_segment = []
-
-        self.windowRange.tau = int(self.Tcoh)  # TYPE UINT4
-        for tstart in self.tboundaries[:-1]:
-            d_detStat = self._get_per_segment_det_stat(tstart)
-            detStat += d_detStat
-            if record_segments:
-                self.detStat_per_segment.append(d_detStat)
+            self.detStat_per_segment = det_stat_per_segment
 
         return detStat
 
-    def _get_per_segment_det_stat(self, tstart):
-        self.windowRange.t0 = int(tstart)  # TYPE UINT4
-
+    def _get_per_segment_det_stat(self):
         FS = lalpulsar.ComputeTransientFstatMap(
-            self.FstatResults.multiFatoms[0], self.windowRange, False
+            self.FstatResults.multiFatoms[0], self._semicoherent_window_trick, False
         )
 
         if self.BSGL is False:
```

Comment:
Instead of the one I suggested at first, please add
```
#FIXME:  BSGL should not be computed per segment, but from summed F!
```

Reply:
I feel we should constraint more the usage of BSGL since neither its implementation nor its results are tested... but that can be done in another PR.